### PR TITLE
Tfenv formula

### PR DESCRIPTION
**Pull request by pyro2927**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR adds a Formula for [tfenv](https://github.com/kamatama41/tfenv).  Checklist above has been completed.

### CODE REVIEW

**Comment by fxcoudert** [comment]
The test does not pass as the exit code is 1. And could you make the test do something more useful than outputting help?

**Comment by pyro2927** [comment]
@fxcoudert, the other pieces of functionality involve hitting an API to return results. Is it safe to make the assumptions that tests will be run with an active internet connection?

**Comment by bfontaine** [comment]
> Is it safe to make the assumptions that tests will be run with an active internet connection?

Yes.

File: Formula/tfenv.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Tfenv < Formula
+  desc "Terraform version manager inspired by rbenv"
+  homepage "https://github.com/kamatama41/tfenv"
+  url "https://github.com/kamatama41/tfenv/archive/v0.4.1.tar.gz"
+  sha256 "f2f7fbd3500d2ae1aef1e2ba51b2f2b81e7cff831c1a63c8a41ade7bce93186b"
+  head "https://github.com/kamatama41/tfenv.git"
+
+  bottle :unneeded
+
+  conflicts_with "terraform", :because => "tfenv symlinks terraform binaries"
```
Comment (fxcoudert) [inline Formula/tfenv.rb]: Please modify `terraform.rb` to add the conflict there too?

**Review by fxcoudert — CHANGES_REQUESTED** [review]
Please squash together the commits related to the same file

File: Formula/tfenv.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    prefix.install ["bin", "libexec"]
+  end
+
+  test do
+    shell_output("#{bin}/tfenv list-remote")
```
Comment (bfontaine) [inline Formula/tfenv.rb]: Please use  `system bin/"tfenv", "list-remote"` if capturing the output is not necessary.

File: Formula/tfenv.rb (hunk 1 above)
Comment (pyro2927) [inline Formula/tfenv.rb]: Conflict added to `terraform.rb` as well.

File: Formula/tfenv.rb (hunk 2 above)
Comment (pyro2927) [inline Formula/tfenv.rb]: Updated.

**Comment by pyro2927** [comment]
@fxcoudert and @bfontaine The requested changes have been made.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Can you make the changes into two commits, one with commit message `tfenv 0.4.1 (new formula)` and the other one with `terraform: add conflict with tfenv`? Thanks

**Comment by pyro2927** [comment]
Commits updated.

**Comment by fxcoudert** [comment]
Nearly there… please amend to fix the following audit errors:

```
tfenv:
  * File should end with a newline
  * C: 19: col 4: Final newline missing.
```

**Comment by pyro2927** [comment]
My mistake, looks like my text editor automatically stripped the newline endings, sorry about that.

**Comment by fxcoudert** [comment]
Thanks @pyro2927 for your contribution to Homebrew!

**Pull request merged**